File: weapon.rb
# frozen_string_literal: true
require_relative 'player'
require_relative 'dice'

module Irrgarten
  class Weapon
    def initialize(power, uses)
      @power = power
      @uses = uses
    end

    def attack
      if @uses > 0
        @uses -= 1
        @power
      end
      0
    end

    def to_s
      "W[" + @power.to_s + "," + @uses.to_s + "]"
    end

    def discard
      Dice.discard_element(@power)
    end
  end
end

File: Weapon.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

/**
 *
 * @author gonzaalovd
 */
public class Weapon extends CombatElement{
    
    public Weapon(float power, int uses){
        super(power,uses);
    }
    
    
    public float attack(){
        return this.produceEffect();
    }
    
    public String toString(){
        return "W" + super.toString();
    }
}
